go/libraries/doltcore/doltdb/root_val.go

```diff
@@ -215,6 +215,45 @@ func (root *RootValue) GenerateTagsForNewColumns(ctx context.Context, tableName
 		})
 	}
 
+	if headRoot != nil {
+		tbl, found, err = headRoot.GetTable(ctx, tableName)
+		if err != nil {
+			return nil, err
+		}
+
+		if found {
+			sch, err := tbl.GetSchema(ctx)
+			if err != nil {
+				return nil, err
+			}
+			var existingColNames []string
+			var existingColTags []uint64
+			_ = sch.GetAllCols().Iter(func(tag uint64, col schema.Column) (stop bool, err error) {
```

Comment:
Probably want a schemasEqual method for this

Reply:
yeah, if we do re-use on replace with slight differences (extra column, drop column, etc) that method wouldn't be used though

Reply:
maybe `schemaOverlap -> map[colName->tag]`? that would be useful now and later